src/debugger/appWorker.ts

```diff
@@ -186,8 +189,12 @@ export class MultipleLifetimesAppWorker {
         socketToApp.on("message",
             (message: any) => this.onMessage(message));
         socketToApp.on("error",
-            (error: Error) => printDebuggingError("An error ocurred while using the socket to communicate with the React Native app", error));
-        return socketToApp;
+            (error: Error) => {
+                deferred.reject(error);
```

Comment:
This looks great for cases when the packager fails to start properly. In cases where the packager falls over later on will we catch it? Does the retry logic on disconnect mean that we end up back in this case anyway?